### PR TITLE
make `podman rmi` more robust

### PR DESCRIPTION
**Pull request by vrothberg**
The c/storage library is subject to TOCTOUs as the central container and
image storage may be shared by many instances of many tools.  As shown
in #6510, it's fairly easy to have multiple instances of Podman running
in parallel and yield image-lookup errors when removing them.

The underlying issue is the TOCTOU of removal being split into multiple
stages of first reading the local images and then removing them.  Some
images may already have been removed in between the two stages. To make
image removal more robust, handle errors at stage two when a given image
is not present (anymore) in the storage.

Fixes: #6510
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

File: test/e2e/rmi_test.go [hunk 1]
```diff
@@ -275,4 +277,31 @@ RUN find $LOCAL
...
+	It("podman image rm - concurrent with shared layers", func() {
+		// #6510 has shown a fairly simple reproducer to force storage
+		// errors during parallel image removal.  Since it's subject to
+		// a race, we may not hit the condition a 100 percent of times
+		// but ocal reproducers hit it all the time.
+
+		var wg sync.WaitGroup
+
+		buildAndRemove := func(i int) {
+			defer wg.Done()
```
Comment (Luap99) [inline test/e2e/rmi_test.go:290]: You also have to add `defer GinkgoRecover()`

Comment (vrothberg) [reply test/e2e/rmi_test.go:291]: Done, thanks.

**Comment by rhatdan** [comment]
LGTM

**Comment by mheon** [comment]
LGTM

**Pull request merged**